pkg/cnab/cnab-to-oci/registry.go

```diff
@@ -188,3 +189,30 @@ func (r *Registry) getDockerClient() (*command.DockerCli, error) {
 	}
 	return cli, nil
 }
+
+func (r *Registry) IsInvocationImageExists(invocationImage string) (bool, error) {
+	ctx := context.Background()
+
+	cli, err := r.getDockerClient()
+	if err != nil {
+		return false, err
+	}
+
+	ref, err := cnab.ParseOCIReference(invocationImage)
+	if err != nil {
+		return false, err
+	}
+
+	imageListOpts := types.ImageListOptions{All: true, Filters: filters.NewArgs(filters.KeyValuePair{Key: "reference", Value: ref.Repository()})}
```

Comment:
```suggestion
	imageListOpts := types.ImageListOptions{All: true, Filters: filters.NewArgs(filters.KeyValuePair{Key: "reference", Value: ref.String()})}
```

ref.Repository will exclude the tag, but we want to know if that tag exactly is present or not in the cache. ref.String will return the full formatted string representing the image.

We could also remove the call to ParseOCIImage I think. And then pass the image name directly to the Value field here. Up to you.

Reply:
I just realized that we can directly use the `invocationImage` name and don't need to use the reference from the `ParseOCIReference` because it yields the same value, thank you!